packages/core/src/mesh/PrimitiveMesh.ts

```diff
@@ -106,48 +109,128 @@ export class PrimitiveMesh {
    * @param width - Cuboid width
    * @param height - Cuboid height
    * @param depth - Cuboid depth
+   * @param accessible - Whether to access data later. If true, you can access the data anytime
    * @returns Cuboid mesh
    */
-  static createCuboid(engine: Engine, width: number = 1, height: number = 1, depth: number = 1): Mesh {
-    const mesh = new Mesh(engine);
+  static createCuboid(
+    engine: Engine,
+    width: number = 1,
+    height: number = 1,
+    depth: number = 1,
+    accessible: boolean = true
+  ): ModelMesh {
+    const mesh = new ModelMesh(engine, "cuboidModelMesh");
 
     const halfWidth: number = width / 2;
     const halfHeight: number = height / 2;
     const halfDepth: number = depth / 2;
 
-    const vertices = new Float32Array(192);
+    const positions: Vector3[] = [];
+    const normals: Vector3[] = [];
+    const uvs: Vector2[] = [];
 
     // prettier-ignore
     // Up
-    vertices[0] = -halfWidth, vertices[1] = halfHeight, vertices[2] = -halfDepth, vertices[3] = 0, vertices[4] = 1, vertices[5] = 0, vertices[6] = 0, vertices[7] = 0,
-    vertices[8] = halfWidth, vertices[9] = halfHeight, vertices[10] = -halfDepth, vertices[11] = 0, vertices[12] = 1, vertices[13] = 0, vertices[14] = 1, vertices[15] = 0,
-    vertices[16] = halfWidth, vertices[17] = halfHeight, vertices[18] = halfDepth, vertices[19] = 0, vertices[20] = 1, vertices[21] = 0, vertices[22] = 1, vertices[23] = 1,
-    vertices[24] = -halfWidth, vertices[25] = halfHeight, vertices[26] = halfDepth, vertices[27] = 0, vertices[28] = 1, vertices[29] = 0, vertices[30] = 0, vertices[31] = 1,
+    positions[0] = new Vector3(-halfWidth, halfHeight, -halfDepth);
+    normals[0] = new Vector3(0, 1, 0);
+    uvs[0] = new Vector2(0, 0);
+
+    positions[1] = new Vector3(halfWidth, halfHeight, -halfDepth);
+    normals[1] = new Vector3(0, 1, 0);
+    uvs[1] = new Vector2(1, 0);
+
+    positions[2] = new Vector3(halfWidth, halfHeight, halfDepth);
+    normals[2] = new Vector3(0, 1, 0);
+    uvs[2] = new Vector2(1, 1);
+
+    positions[3] = new Vector3(-halfWidth, halfHeight, halfDepth);
+    normals[3] = new Vector3(0, 1, 0);
+    uvs[3] = new Vector2(0, 1);
```

Comment:
You can write like before with "prettier-ignore":
positions[0] = new Vector3(-halfWidth, halfHeight, -halfDepth),normals[0] = new Vector3(0, 1, 0),uvs[0] = new Vector2(0, 0);

If you write like this you can delete "prettier-ignore".